src/spyglass/image.cr

```diff
@@ -0,0 +1,486 @@
+    def draw_line(start : LibSod::SodPts, stop : LibSod::SodPts, rgb : RGBColor)
+      tmp = copy
+      LibSod.sod_image_draw_line(tmp, start.to_i, stop.to_i, rgb[0].to_f, rgb[1].to_f, rgb[2].to_f)
+      Image.new(tmp)
+    end
+
+    # Draw a single circle via the Midpoint Circle Algorithm on an input image. This function is very
+    # useful when working with the CNN/Realnet interfaces for example to mark a region of
+    # interest (i.e. face) coordinates.
+    def draw_circle(x : Number, y : Number, radius : Number, rgb : RGBColor)
+      tmp = copy
+      LibSod.sod_image_draw_circle(tmp, x.to_i, y.to_i, radius.to_i, rgb[0].to_f, rgb[1].to_f, rgb[2].to_f)
+      Image.new(tmp)
+    end
+
+    # Draw a single circle with a certain thickness via the Midpoint Circle Algorithm. See `Image.draw_circle`.
+    def draw_circle(x : Number, y : Number, radius : Number, width : Number, rgb : RGBColor)
+      tmp = copy
+      LibSod.sod_image_draw_circle(tmp, x.to_i, y.to_i, radius.to_i, width.to_i, rgb[0].to_f, rgb[1].to_f, rgb[2].to_f)
+      Image.new(tmp)
+    end
+
+    # Scale an input image to the desired dimension.
+    def resize(width : Number, height : Number)
+      tmp = copy
+      LibSod.sod_resize_image(tmp, width.to_i, height.to_i)
+      Image.new(tmp)
+    end
+
+    # Scale an input image to the maximum available dimension (i.e. Pick the maximum value
+    # from the image width or height). For standard image resizing, please rely on
+    # `Image#resize` instead.
+    def resize_max(max : Number)
+      tmp = copy
+      LibSod.sod_resize_max(tmp, max.to_i)
+      Image.new(tmp)
+    end
+
+    # Scale an input image to the minimum available dimension (i.e. Pick the minimum value
+    # from the image width or height). For standard image resizing, please rely on
+    # `Image#resize` instead.
+    def resize_min(min : Number)
+      tmp = copy
+      LibSod.sod_resize_min(tmp, min.to_i)
+      Image.new(tmp)
+    end
+
+    # Extract a random region from an input image. For standard image cropping, you should
+    # rely on `Image#crop` instead.
+    def random_crop(width : Number, height : Number)
+      tmp = copy
+      LibSod.sod_random_crop_image(tmp, width.to_i, height.to_i)
+      Image.new(tmp)
+    end
+
+    # Perform small perturbations in scale and position to an input image. For standard image
+    # cropping, rotation & scaling you should rely respectively on `Image#crop`,
+    # `Image#rotate` and `Image#scale`.
+    def random_augment(angle : Number, aspect : Number, low : Number, high : Number, size : Number)
+      tmp = copy
+      LibSod.sod_random_augment_image(tmp, angle.to_f, aspect.to_f, low.to_i, high.to_i, size.to_i)
+      Image.new(tmp)
+    end
+
+    # Perform constant value addition on each pixel of a given image (Image Arithmetic). If you
+    # want to process a single channel, then call `Image.translate_channel` instead.
+    def translate(s : Float64)
+      unless (0.0..1.0).includes?(s)
```

Comment:
This check should be added as a method, it's done in a number of other functions.

Reply:
Agreed